src/devices/tuya.ts

```diff
@@ -10600,6 +10600,113 @@ const definitions: Definition[] = [
+            e.numeric('external_temperature_input', ea.STATE).withLabel('Floor temperature').withUnit('°C')
+                .withDescription('Temperature from floor sensor'),
+            e.numeric('brightness', ea.STATE_SET).withValueMin(0).withValueMax(8).withValueStep(1)
+                .withLabel('Screen brightness 06:00 - 22:00')
+                .withDescription('0 - on for 10 seconds. Default: 6'),
+            e.numeric('display_brightness', ea.STATE_SET).withValueMin(0).withValueMax(8).withValueStep(1)
+                .withLabel('Screen brightness 22:00 - 06:00')
+                .withDescription('0 - on for 10 seconds. Default: 3'),
+            e.text('schedule_monday', ea.STATE_SET).withLabel('Schedule for monday')
+                .withDescription('Default: 06:00/20.0 11:30/20.0 13:30/20.0 17:30/20.0'),
+            e.text('schedule_tuesday', ea.STATE_SET).withLabel('Schedule for tuesday')
+                .withDescription('Default: 06:00/20.0 11:30/20.0 13:30/20.0 17:30/20.0'),
+            e.text('schedule_wednesday', ea.STATE_SET).withLabel('Schedule for wednesday')
+                .withDescription('Default: 06:00/20.0 11:30/20.0 13:30/20.0 17:30/20.0'),
+            e.text('schedule_thursday', ea.STATE_SET).withLabel('Schedule for thursday')
+                .withDescription('Default: 06:00/20.0 11:30/20.0 13:30/20.0 17:30/20.0'),
+            e.text('schedule_friday', ea.STATE_SET).withLabel('Schedule for friday')
+                .withDescription('Default: 06:00/20.0 11:30/20.0 13:30/20.0 17:30/20.0'),
+            e.text('schedule_saturday', ea.STATE_SET).withLabel('Schedule for saturday')
+                .withDescription('Default: 06:00/20.0 11:30/20.0 13:30/20.0 17:30/20.0'),
+            e.text('schedule_sunday', ea.STATE_SET).withLabel('Schedule for sunday')
+                .withDescription('Default: 06:00/20.0 11:30/20.0 13:30/20.0 17:30/20.0'),
+            e.enum('factory_reset', ea.STATE_SET, ['factory reset'])
+                .withLabel('Factory reset').withDescription('Reset all settings to factory ones'),
+        ],
+        meta: {
+            tuyaDatapoints: [
+                [1, 'system_mode', tuya.valueConverterBasic.lookup((_, device) => {
+                    return {'OFF': false, 'HEAT': true};
+                })],
+                [2, 'preset', tuya.valueConverterBasic.lookup((_, device) => {
+                    return {'AUTO': tuya.enum(0), 'MANUAL': tuya.enum(1)};
+                })],
+                [16, 'current_heating_setpoint', tuya.valueConverter.divideBy10],
+                [19, 'max_temperature', tuya.valueConverter.divideBy10],
+                [24, 'local_temperature', tuya.valueConverter.divideBy10],
+                [26, 'min_temperature', tuya.valueConverter.divideBy10],
+                [27, 'local_temperature_calibration', tuya.valueConverter.raw],
+                [28, 'factory_reset', tuya.valueConverterBasic.lookup((_, device) => {
```

Comment:
```js
[28, 'factory_reset', tuya.valueConverterBasic.lookup({'factory_reset': true}],
```

Please use this, also for the other `tuya.valueConverterBasic.lookup`

Reply:
Thank you for this. I'm not a programmer, but I'll try.
Fixed.